packages/web/components/one-click-trading/one-click-trading-settings.tsx

```diff
@@ -0,0 +1,244 @@
+import { Dec, PricePretty } from "@keplr-wallet/unit";
+import classNames from "classnames";
+import dayjs from "dayjs";
+import Image from "next/image";
+import React from "react";
+import { useState } from "react";
+
+import { Icon } from "~/components/assets";
+import { Button, buttonCVA } from "~/components/buttons";
+import IconButton from "~/components/buttons/icon-button";
+import { Switch } from "~/components/control";
+import {
+  Screen,
+  ScreenGoBackButton,
+  ScreenManager,
+} from "~/components/screen-manager";
+import { useTranslation } from "~/hooks";
+import { DEFAULT_VS_CURRENCY } from "~/server/queries/complex/assets/config";
+
+type Classes = "root";
+
+enum SettingsScreens {
+  Main = "main",
+  SpendLimit = "spendLimit",
+  NetworkFeeLimit = "networkFeeLimit",
+  SessionPeriod = "sessionPeriod",
+}
+
+interface OneClickTradingSettingsProps {
+  classes?: Partial<Record<Classes, string>>;
+  onClose?: () => void;
+}
+
+const OneClickTradingSettings = ({
+  classes,
+  onClose,
+}: OneClickTradingSettingsProps) => {
+  const { t } = useTranslation();
+
+  const [parameters, setParameters] = useState({
+    isOneClickEnabled: false,
+    spendLimit: new PricePretty(DEFAULT_VS_CURRENCY, new Dec(5000)),
+    networkFeeLimit: new Dec(0.000708),
+    sessionPeriod: dayjs.duration(1, "hour"),
+  });
+
+  const isDisabled = !parameters.isOneClickEnabled;
+
+  const screenGoBackButton = (
+    <ScreenGoBackButton className="absolute top-7 left-7" />
+  );
+
+  return (
+    <ScreenManager defaultScreen={SettingsScreens.Main}>
+      {({ setCurrentScreen }) => (
+        <>
+          <Screen screenName="main">
+            <div className={classNames("flex flex-col gap-6", classes?.root)}>
+              <IconButton
+                onClick={onClose}
+                className="absolute top-7 left-7 w-fit text-osmoverse-400 hover:text-osmoverse-100"
+                icon={<Icon id="chevron-left" width={16} height={16} />}
+                aria-label="Go Back"
+                mode="unstyled"
+              />
+
+              <h1 className="w-full text-center text-h6 font-h6 tracking-wider">
+                {t("oneClickTrading.settings.header")}
+              </h1>
+
+              <div className="px-8">
+                <div className="flex gap-4 rounded-xl bg-osmoverse-825 px-4 py-3">
+                  <Image
+                    src="/images/1ct-mobile-icon.svg"
+                    alt="1ct icon"
+                    width={32}
+                    height={32}
+                    className="self-start"
+                  />
+                  <p className="text-body2 font-body2 text-osmoverse-300">
+                    {t("oneClickTrading.settings.description")}{" "}
+                    <a
+                      className={buttonCVA({
+                        mode: "text",
+                        className: "!inline w-auto px-0 text-body2 font-body2",
```

Comment:
Why the `!`?

Reply:
We need it to override the default `display` of ButtonCVA